src/AppBundle/Entity/SiteRepository.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\EntityRepository;
+
+class SiteRepository extends EntityRepository
+{
+    /**
+     * @param string $name
+     * @param Framework $framework
+     * @return array
+     */
+    public function search( $name = null, Framework $framework = null)
+    {
+
+        $qb = $this->createQueryBuilder('s');
+        if ($name) {
+            $qb->andWhere('s.name LIKE :name')
+                ->setParameter('name', "%" . $name . "%");
```

Comment:
you need to add slashes to special characters when using `LIKE`

     >setParameter('name', '%' . addcslashes($name, '%_') . '%');